### ISSUE TITLE
VM crush

### ISSUE THREAD

**Issue by vorov2** [vorov2/dyalect#369]
The following code causes VM to crush:

```swift
type Box(key)
const boxCollector = []
func Box.dispose() {
    boxCollector.add(this:key)
}

func doit() {
    var x = {
        throw 24
        auto b = Box(12)
    }
}
try {
    doit()
} catch e {
    auto b = Box(e)
}
assert([24], boxCollector)

```

**Comment by vorov2** [comment]
Related with new `auto` feature.
Problem 1: Run-time tries to `dispose` a variable that is not yet initialized.

**Comment by vorov2** [comment]
Problem 2 resolves as #370
